### PR TITLE
Update Vector.java

### PR DESCRIPTION
**Pull request by Junjian123**
Another attempt @rivlev

### CODE REVIEW

**Comment by Junjian123** [comment]
I updated my Vector class which includes my own testing code in main class @rivlev

**Comment by Junjian123** [comment]
@rivlev Professor, can you see this pull request?

File: Vector.java [hunk 1]
```diff
@@ -51,21 +51,51 @@ public E set(int index, E element) {
...
    	if (size == data.length) grow(); 
-    data[size++] = element;
+    data[size] = element;
+    size++;
     return true;
   }
   
   public void add(int index, E element) {
-		// Add element at index.
+	if(size++==data.length) grow();	// insert the given element into index, and shift the elements after index.
```
Comment (rivlev) [inline Vector.java:60]: check index -5

File: Vector.java [hunk 2]
```diff
@@ -51,21 +51,51 @@ public E set(int index, E element) {
...
+        data[index]=element;
+        
+        for(int i=0;i<size;i++)
+            System.out.print(data[i]+" ");
   }
   
   public E remove(int index) {
-		// Remove the element at index. Make sure there are no gaps
-		// Return the removed element
+        System.out.println("the size before remove is "+size);
```
Comment (rivlev) [inline Vector.java:71]: check index -5

File: Vector.java [hunk 3]
```diff
@@ -51,21 +51,51 @@ public E set(int index, E element) {
...
+        System.out.println(" the order before change is ");
+        for(int i=0;i<size;i++)
+            System.out.print(data[i]+" ");
+        System.out.println();
+	E removed=data(index);	// Remove the element at index. Make sure there are no gaps
+        //data[index]=null;
+        if(size++==data.length) grow();
```
Comment (rivlev) [inline Vector.java:78]: Why are you incrementing size? Nothing is being added here. 
-3

File: Vector.java [hunk 4]
```diff
@@ -51,21 +51,51 @@ public E set(int index, E element) {
...
+        for(int i=index;i<size;i++){
```
Comment (rivlev) [inline Vector.java:79]: This should end at size-2
-4

**Review by rivlev — COMMENTED** [review]
83

**Pull request closed without merging**